### PR TITLE
Add auth console command for web console access

### PR DESCRIPTION
**Pull request by osterman**
## what
- Add `atmos auth console` command to open cloud provider web consoles using authenticated credentials
- Implement AWS console access via federation endpoint (similar to aws-vault login)
- Add 100+ AWS service destination aliases for convenient access
- Create dedicated `pkg/http` package for HTTP client utilities
- Add pretty formatted output using lipgloss with Atmos theme colors
- Consolidate browser opening functionality to existing `OpenUrl` helper

## why
- Provides convenient browser access to cloud consoles without manually copying credentials
- Eliminates context switching between terminal and browser for console access
- Uses provider-native federation endpoints for secure temporary access
- Extensible interface pattern supports future Azure/GCP implementations

## features
- **Service Aliases**: Use shorthand like `s3`, `ec2`, `lambda` instead of full console URLs
- **Autocomplete**: Shell completion for destination and identity flags  
- **Session Control**: Configurable duration (up to 12 hours for AWS) with expiration display
- **Clean Output**: URL only shown on error or with `--no-open` flag
- **Scriptable**: `--print-only` flag for piping URLs to other tools
- **Provider-Agnostic**: Interface design ready for multi-cloud support

## implementation
- Created `ConsoleAccessProvider` interface in `pkg/auth/types/interfaces.go`
- Implemented `ConsoleURLGenerator` for AWS using federation endpoint
- Added `ResolveDestination()` with case-insensitive alias lookup
- Moved HTTP utilities from `pkg/utils` to dedicated `pkg/http` package
- Used existing `OpenUrl()` function for cross-platform browser opening
- Added comprehensive tests achieving 85.9% coverage

## testing
- Unit tests for console URL generation (15 test cases)
- Unit tests for destination alias resolution (100+ aliases tested)
- Mock HTTP client for testing without network calls
- Table-driven tests with edge case coverage

## documentation
- CLI reference: `website/docs/cli/commands/auth/console.mdx`
- Blog post: `website/blog/2025-10-20-auth-console-web-access.md`
- Proposal document: `docs/proposals/auth-web-console.md`
- Embedded markdown usage examples

## references
- Similar to aws-vault's console login feature
- AWS Federation Endpoint: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_enable-console-custom-url.html

## Summary by CodeRabbit

* **New Features**
  * Added atmos auth console: opens cloud provider web consoles via temporary sign-in URLs (AWS supported now; Azure/GCP planned).
  * Supports service aliases (s3, ec2, etc.), full destination URLs, session duration (AWS up to 12h), issuer, --print-only, --no-open and identity selection/completion.

* **Documentation**
  * New CLI docs, usage guide, PRD and blog post with examples and troubleshooting.

* **Tests**
  * Expanded tests and CI snapshots for the new command and destination resolution.

### CODE REVIEW

File: cmd/auth_console.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+}
+
+func executeAuthConsoleCommand(cmd *cobra.Command, args []string) error {
+	handleHelpRequest(cmd, args)
+
+	// Load atmos config.
+	atmosConfig, err := cfg.InitCliConfig(schema.ConfigAndStacksInfo{}, false)
+	if err != nil {
+		return fmt.Errorf("failed to load atmos config: %w", err)
+	}
```
[the remark this answers is not in the document]
Comment (osterman) [reply cmd/auth_console.go]: ✅ **Resolved** - All error wrapping has been implemented correctly:

**Sentinel Error Created:**
- `ErrAuthConsole` defined in `errors/errors.go` (commit dbb1fc367)

**All Locations Updated with Sentinel Wrapping:**
- Line 79: authentication failed - wraps with ErrAuthConsole
- Line 91: console provider error - wraps with ErrAuthConsole  
- Line 104: generate console URL - wraps with ErrAuthConsole
- Line 206: load atmos config - wraps with ErrAuthConsole (initializeAuthManager)
- Line 211: create auth manager - wraps with ErrAuthConsole (initializeAuthManager)
- Line 226: get default identity - wraps with ErrAuthConsole (resolveIdentityName)
- **Line 230: no default identity guard** - wraps with ErrAuthConsole ✅
- Line 245: retrieve credentials from store - wraps with ErrAuthConsole (retrieveCredentials)
- Line 249: no credentials available - wraps with ErrAuthConsole (retrieveCredentials)

**Error Chain Preservation:**
All errors use `fmt.Errorf("%w: ...: %w", errUtils.ErrAuthConsole, err)` pattern:
- Sentinel error wrapped with `%w` (first position) - testable with `errors.Is()`
- Underlying error wrapped with `%w` (second position) - preserves original error chain
- Callers can use `errors.Is(err, errUtils.ErrAuthConsole)` to detect auth console errors
- Callers can unwrap to inspect the underlying cause

**Lines 65, 71, 84:** These return errors from helper functions that already wrap with ErrAuthConsole internally, so no double-wrapping needed.

File: cmd/auth_console.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+		return generator, nil
+	case types.ProviderKindAzureOIDC:
+		return nil, fmt.Errorf("%w: Azure console access not yet implemented (coming soon)", errUtils.ErrProviderNotSupported)
+	case types.ProviderKindGCPOIDC:
+		return nil, fmt.Errorf("%w: GCP console access not yet implemented (coming soon)", errUtils.ErrProviderNotSupported)
+	default:
+		return nil, fmt.Errorf("%w: provider %q does not support web console access", errUtils.ErrProviderNotSupported, providerKind)
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (osterman) [reply cmd/auth_console.go:203]: ✅ **Resolved** - Added perf.Track() to all critical helper functions:

- `getConsoleProvider` (line 181)
- `initializeAuthManager` (line 206)
- `resolveIdentityName` (line 223)
- `retrieveCredentials` (line 244)

All functions now follow the mandatory pattern with `defer perf.Track(nil, "cmd.functionName")()` and blank line after.

Commit: 241339d09

File: go.mod [hunk 3]
```diff
@@ -42,6 +42,7 @@ require (
 	github.com/gobwas/glob v0.2.3
 	github.com/goccy/go-yaml v1.18.0
 	github.com/gofrs/flock v0.13.0
+	github.com/golang/mock v1.1.1
```
[the remark this answers is not in the document]
Comment (osterman) [reply go.mod]: ✅ **Resolved** - Removed legacy gomock dependency:

- Removed `github.com/golang/mock v1.1.1` from go.mod
- Updated all imports to use `go.uber.org/mock/gomock`
- Ran `go mod tidy` to clean up dependencies
- Project now only uses `go.uber.org/mock v0.6.0`

Commit: 241339d09

File: pkg/http/mock_client.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+// Package http is a generated GoMock package.
+package http
+
+import (
+	http "net/http"
+	reflect "reflect"
+
+	gomock "github.com/golang/mock/gomock"
+)
```
[the remark this answers is not in the document]
Comment (osterman) [reply pkg/http/mock_client.go]: ✅ **Resolved** - Fixed mock_client.go import path:

- Updated import from `github.com/golang/mock/gomock` to `go.uber.org/mock/gomock`
- Regenerated mock using `mockgen` with new import path
- Removed incompatible `T.Helper()` calls from generated code
- All tests passing with updated mock

Commit: 241339d09

File: cmd/auth_login.go [hunk 5]
```diff
@@ -135,6 +135,9 @@ func displayAuthSuccess(whoami *authTypes.WhoamiInfo) {
 	}
 
 	// Create minimal charmbracelet table.
+	// Note: Padding variation across platforms was causing snapshot test failures.
+	// The table auto-sizes columns but the final width varied (Linux: 40 chars, macOS: 45 chars).
+	// Removed `.Width()` constraint as it was causing word-wrapping issues.
 	t := table.New().
```
[the remark this answers is not in the document]
Comment (aknysh) [reply cmd/auth_login.go:141]: ❌ Comment is PARTIALLY INCORRECT and conflates different use cases

  Actual .Width() Usage in Codebase:

  ./cmd/auth_console.go:145        # labelStyle.Width(consoleLabelWidth)
  ./internal/tui/atmos/column.go   # Full-screen TUI
  ./internal/tui/workflow/column.go # Full-screen TUI  
  ./pkg/ui/heatmap/heatmap.go      # Full-screen TUI

  Files Incorrectly Listed in Comment:

  - ❌ internal/exec/vendor_model.go - Uses lipgloss.Width() function (measures width), not .Width() method (sets width)
  - ❌ pkg/list/format/table.go - No .Width() usage
  - ❌ cmd/version/formatters.go - No .Width() usage
  - ❌ cmd/auth_whoami.go - No .Width() usage

  Context Review:

  auth_login.go removal was specifically for:
  - A charmbracelet table that had cross-platform padding issues
  - Causing snapshot test failures (Linux: 40 chars, macOS: 45 chars)
  - Word-wrapping problems with fixed width

  auth_console.go usage is DIFFERENT:
  - Uses .Width() on a label style, not a table
  - Fixed width (18 chars) ensures labels align in key-value output
  - No snapshot tests exist for auth_console output
  - Tests only check "doesn't panic", not exact formatting

  My Recommendation:

  REJECT THIS COMMENT because:

  1. Different use cases require different approaches:
    - Full-screen TUI apps (heatmap, workflow, atmos) → MUST have width constraints for proper rendering
    - Simple console output (auth_console labels) → Fixed width OK for alignment, no snapshot concerns
    - Tables (auth_login) → Auto-sizing better for cross-platform consistency
  2. auth_console.go is fine as-is:
    - Label width ensures clean alignment: "Provider:        aws"
    - No snapshot tests to break
    - Not causing word-wrap issues
  3. Standardization would be wrong here - these are legitimately different scenarios with different requirements

**Comment by aknysh** [comment]
@coderabbitai full review
